lib/ext/heroku/command/addons.rb

```diff
@@ -56,30 +56,54 @@ def index
     # addons:plans
     #
     # list all available add-on plans
+    def services
+      addon_services = api.request(
+        :expects  => [200, 206],
+        :headers  => { "Accept" => "application/vnd.heroku+json; version=3" },
+        :method   => :get,
+        :path     => "/addon-services"
+      ).body
+
+      display_table(addon_services, %w[name human_name], %w[Slug Name])
+      display "\nSee plans with `heroku addons:plans SERVICE`"
+    end
+
+    alias_command "addons:list", "addons:services"
+
+    # addons:plans SERVICE
     #
-    # --region REGION      # specify a region for add-on plan availability
-    #
-    #Example:
-    #
-    # $ heroku addons:plans --region eu
-    # === available
-    # adept-scale:battleship, corvette...
-    # adminium:enterprise, petproject...
-    #
+    # list all available plans for an add-on service
     def plans
-      addons = heroku.addons(options)
-      if addons.empty?
-        display "No add-ons available currently"
-      else
-        partitioned_addons = partition_addons(addons)
-        partitioned_addons.each do |key, addons|
-          partitioned_addons[key] = format_for_display(addons)
-        end
-        display_object(partitioned_addons)
+      service = args.shift
+      raise CommandFailed.new("Missing add-on service") if service.nil?
+
+      service = api.request(
+        :expects  => [200, 206],
+        :headers  => { "Accept" => "application/vnd.heroku+json; version=3" },
+        :method   => :get,
+        :path     => "/addon-services/#{service}"
+      ).body
+
+      display_header("#{service['human_name']} Plans")
+
+      plans = api.request(
+        :expects  => [200, 206],
+        :headers  => { "Accept" => "application/vnd.heroku+json; version=3" },
+        :method   => :get,
+        :path     => "/addon-services/#{service['id']}/plans"
+      ).body
+
+      plans = plans.sort_by { |p| [(!p['default']).to_s, p['price']['cents']] }.map do |plan|
```

Comment:
I think everything in API is sorted by id and or name so that the pagination works possibly. That being said, it's probably customisable per endpoint but I haven't looked into it.